build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -241,9 +241,7 @@ class AssetGraph {
     // Pre-emptively compute digests for the new and modified nodes we know have
     // outputs.
     await _setLastKnownDigests(
-        newAndModifiedNodes
-            .where((node) => node.isValidInput && node.outputs.isNotEmpty),
-        digestReader);
+        newAndModifiedNodes.where((node) => node.isValidInput), digestReader);
```

Comment:
We should make sure this doesn't hurt performance significantly but I don't think it really will, unless somebody has a ton of static files (but even if they are ever served we would want to do this).

The other option would be to only set the digest if it previously had a digest set.

Reply:
I think in general we should only end up here if it did have a digest set, but we could also be more explicit about it.